modules/platform.py

```diff
@@ -1253,28 +1201,614 @@ def get_lockdown(self, **kwargs):
             params=_generate_params(kwargs, valid_keys, deprecated_keys),
         )
 
+    def download_vectra_ruleset(self, filename=None):
+        if filename is None:
+            filename = "curated.rules"
+        elif not isinstance(filename, str):
+            filename = "curated.rules"
+            raise TypeError(
+                "Filename must be of type str. File is being named 'curated.rules'."
+            )
+
+        p = Path(filename)
+        p.parent.mkdir(parents=True, exist_ok=True)
+
+        resp = self._request(
+            method="get", url=self.url + "/vectra-match/download-vectra-ruleset"
+        )
+
+        resp = requests.get(url=resp.json()["download_url"])
+
+        with open(str(filename), "wb") as file:
+            for chunk in resp.iter_content(chunk_size=8192):
+                if chunk:
+                    file.write(chunk)
+        return resp
+
+    def upload_match_ruleset(self, **kwargs):
+        """
+        Upload vectra-match rules
+        :param file: name of ruleset desired to be uploaded (required)
+        :param notes: notes about the uploaded file (optional)
+        """
+        file_path = kwargs.get("file_path", False)
+        if not file_path:
+            raise ValueError("A ruleset filename is required.")
+        notes = kwargs.get("notes", "")
+        headers = {"Authorization": self.headers["Authorization"]}
+
+        # Get the upload url
+        resp = self._request(
+            method="post",
+            url=f"{self.url}/vectra-match/rules/upload/",
+            headers=headers,
+            json={"file_name": file_path, "notes": notes},
+        )
+
+        upload_url = resp.json()["urls"][0]
+        upload_id = resp.json()["id"]
+
+        # Upload the file to the provided url
+        payload = open(f"{file_path}", "rb")
+        resp = requests.put(upload_url, data=payload)
+
+        if resp.status_code == 200:
```

Comment:
Minor: may be more future proof if you use [request.codes.ok ](https://requests.readthedocs.io/en/latest/user/quickstart/#response-status-codes).  That would handle if you get codes like 200 and 204